packages/lit-html/src/directives/class-map.ts

```diff
@@ -42,17 +43,30 @@ class ClassMapDirective extends Directive {
   }
 
   render(classInfo: ClassInfo) {
-    return Object.keys(classInfo)
-      .filter((key) => classInfo[key])
-      .join(' ');
+    // Add spaces to ensure separation from static classes
+    return (
+      ' ' +
+      Object.keys(classInfo)
+        .filter((key) => classInfo[key])
+        .join(' ') +
+      ' '
+    );
   }
 
   override update(part: AttributePart, [classInfo]: DirectiveParameters<this>) {
     // Remember dynamic classes on the first render
     if (this._previousClasses === undefined) {
       this._previousClasses = new Set();
+      if (part.strings !== undefined) {
```

Comment:
Based on the restrictions that (1) using `classMap` means you can only have 1 dynamic part in the attribute, and (2) attribute bindings set the entire attribute value, this change should make `classMap` do the equivalent of emitting a string of the truthy keys.